src/iam.ts

```diff
@@ -262,14 +263,28 @@ export class IAM extends IAMBase {
   async updateDidDocument(options: {
     didAttribute: DIDAttribute;
     data: IUpdateData;
+    did?: string;
     validity?: number;
   }): Promise<boolean> {
-    const { didAttribute, data, validity } = options;
-    if (this._document) {
+    const { didAttribute, data, validity, did } = options;
+
+    if (!this._document) {
+      throw new Error(ERROR_MESSAGES.DID_DOCUMENT_NOT_INITIALIZED);
+    }
+
+    if (!this._didSigner) {
+      throw new Error(ERROR_MESSAGES.SIGNER_NOT_INITIALIZED);
+    }
+
+    if (!did) {
       const updated = await this._document.update(didAttribute, data, validity);
       return Boolean(updated);
     }
-    throw new Error(ERROR_MESSAGES.DID_DOCUMENT_NOT_INITIALIZED);
+
+    const operator = new ProxyOperator(this._didSigner, this._registrySetting, addressOf(did));
+    const update = await operator.update(did, DIDAttribute.PublicKey, data);
```

Comment:
I don't think that it will always be a `PublicKey`. Should use the `didAttribute` param